### PR TITLE
Add support for KEDA HPA config to Helm Chart

### LINKED ISSUE #24103: Add support for KEDA HPA Config to Helm Chart
### Description

> When managing the scale of a group of replicas using the HorizontalPodAutoscaler, it is possible that the number of replicas keeps fluctuating frequently due to the dynamic nature of the metrics evaluated. This is sometimes referred to as thrashing, or flapping.
https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#flapping

Sometimes clusters need to restrict the flapping of Airflow worker replicas.
KEDA supports [`advanced.horizontalPodAutoscalerConfig`](https://keda.sh/docs/1.4/concepts/scaling-deployments/).
It would be great if the users would have the option in the helm chart to configure scale down behavior.

### Use case/motivation

KEDA currently cannot set advanced options.
We want to set advanced options like `scaleDown.stabilizationWindowSeconds`, `scaleDown.policies`.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Swalloow**
Add support for KEDA HPA config to Helm Chart
Closes: #24103 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by llilyy** [comment]
Any updates on this PR?

File: tests/charts/test_keda.py [hunk 1]
```diff
@@ -52,6 +52,50 @@ def test_keda_enabled(self, executor, is_created):
...
+                                "behavior": {
+                                    "scaleDown": {
+                                        "stabilizationWindowSeconds": 300,
+                                        "policies": [{"type": "Percent", "value": 100, "periodSeconds": 15}],
+                                    }
+                                }
+                            }
+                        },
+                    },
+                    "persistence": {"enabled": False},
```
Comment (jedcunningham) [inline tests/charts/test_keda.py]:
```suggestion
```

We don't need this, right?

File: tests/charts/test_keda.py [hunk 2]
```diff
@@ -52,6 +52,50 @@ def test_keda_enabled(self, executor, is_created):
...
+                },
+                'executor': executor,
```
Comment (jedcunningham) [inline tests/charts/test_keda.py]:
```suggestion
                "executor": executor,
```

Nit: lets keep consistent quotes.

File: tests/charts/test_keda.py [hunk 3]
```diff
@@ -52,6 +52,50 @@ def test_keda_enabled(self, executor, is_created):
...
+                        "advanced": {
+                            "horizontalPodAutoscalerConfig": {
+                                "behavior": {
+                                    "scaleDown": {
+                                        "stabilizationWindowSeconds": 300,
+                                        "policies": [{"type": "Percent", "value": 100, "periodSeconds": 15}],
+                                    }
+                                }
+                            }
+                        },
```
Comment (jedcunningham) [inline tests/charts/test_keda.py]:
```suggestion
                        "advanced": expected_advanced,
```

Might as well cut down on duplication.

File: tests/charts/test_keda.py (hunk 1 above)
Comment (Swalloow) [reply tests/charts/test_keda.py]: Thanks, I fixed it.

File: tests/charts/test_keda.py (hunk 2 above)
Comment (Swalloow) [reply tests/charts/test_keda.py]: Thanks, I fixed it.

File: tests/charts/test_keda.py (hunk 3 above)
Comment (Swalloow) [reply tests/charts/test_keda.py]: Thanks, I fixed it.

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**